api/lovelace/models.py

```diff
@@ -32,3 +60,63 @@ class User:
     updated_at: Mapped[datetime] = mapped_column(
         init=False, server_default=func.now(), onupdate=func.now()
     )
+    sumbissions: Mapped[list['Submission']] = relationship(
+        init=False, back_populates='user', cascade='all, delete-orphan'
+    )
+
+
+@table_registry.mapped_as_dataclass
+class Problem:
+    __tablename__ = 'problems'
+
+    id: Mapped[int] = mapped_column(init=False, primary_key=True)
+    name: Mapped[str] = mapped_column(unique=True)
+    description: Mapped[str]
+    problem_input: Mapped[str]
+    problem_output: Mapped[str]
+    sumbissions: Mapped[list['Submission']] = relationship(
+        init=False, back_populates='user', cascade='all, delete-orphan'
+    )
+    difficulty: Mapped[Difficulty]
+    status: Mapped[ProblemStatus] = mapped_column(
+        default=ProblemStatus.todo, server_default='todo'
+    )
+
+
+@table_registry.mapped_as_dataclass
+class Submission:
+    __tablename__ = 'submissions'
+
+    id: Mapped[int] = mapped_column(init=False, primary_key=True)
+    user_id: Mapped[int] = mapped_column(ForeignKey('users.id'))
+    user: Mapped[User] = relationship(init=False, back_populates='submissions')
+    problem_id: Mapped[int] = mapped_column(ForeignKey('problems.id'))
+    problem: Mapped[Problem] = relationship(
+        init=False, back_populates='submissions'
+    )
+    body: Mapped[str]
+    created_at: Mapped[datetime] = mapped_column(
+        init=False, server_default=func.now()
+    )
+    language: Mapped[str]
+    status: Mapped[SubmissonStatus]
+
+
+@table_registry.mapped_as_dataclass
+class Tag:
+    __tablename__ = 'tags'
+
+    id: Mapped[int] = mapped_column(init=False, primary_key=True)
+    name: Mapped[str] = mapped_column(unique=True)
+
+
+@table_registry.mapped_as_dataclass
+class ProblemsTags:
+    __tablename__ = 'problems_tags'
+
+    problem_id: Mapped[int] = mapped_column(primary_key=True)
+    tag_id: Mapped[int] = mapped_column(primary_key=True)
+    problem: Mapped[Problem] = relationship(init=False)
+    tag: Mapped[Tag] = relationship(init=False)
+
+    __table_args__ = (PrimaryKeyConstraint('problem_id', 'tag_id'),)
```

Comment:
Tava pesquisando aqui e o jeito de fazer many-to-many é diferente. Não precisa criar uma nova classe, o cara cria um associated_table.